IaC/cloud.cd/JenkinsStack.yml

```diff
@@ -742,6 +762,26 @@ Resources:
                   fi
               }
 
+              setup_ssh_keys() {
+                KEYS_LIST="mykola.marzhan evgeniy.patlan slava.sarzhan illia.pshonkin alex.miroshnychenko eduardo.casarero santiago.ruiz andrew.siemen kamil.babayev"
+
+                for KEY in $KEYS_LIST; do
+                    if [[ ! -f "/tmp/$KEY.pub" ]]; then
+                        STATUS=$(curl -Is https://www.percona.com/get/engineer/KEY/$KEY.pub | head -n1 | awk '{print $2}')
+                        if [[ ${STATUS} -eq 200 ]]; then
+                            until wget -O "/tmp/$KEY.pub" "https://www.percona.com/get/engineer/KEY/$KEY.pub"; do
+                                echo "sleep before retry"
+                                sleep 1
+                            done
+                        elif [[ ${STATUS} -eq 404 ]]; then
+                            echo "Skipping key $KEY"
+                        fi
+                    fi
+                    cat /tmp/$KEY.pub | tee -a /home/ec2-user/.ssh/authorized_keys
+                    rm /tmp/$KEY.pub
```

Comment:
I think these commands will fail if `${STATUS} -eq 404`
we have `set -o errexit` above

Reply:
You're right, I'll add flag

Reply:
Reworked in 01bac22